### DISCUSSION TITLE
DataGrid always showing a row selected after populating

### DISCUSSION THREAD

**Discussion by TangDave** [AvaloniaUI/Avalonia#16965] [Q&A] [answered]
After populating my DataGrid, I'm trying to have no row selected by having:
`grdHistory.SelectedIndex = -1;`

However, this seems to have no effect - a row is always showing as selected. What am I missing?

This seems to occur if the grid is sorted after populating - e.g.,
`grdHistory.Columns[0].Sort(ListSortDirection.Ascending);`

**Comment by timunie** [comment]
Not sure without seeing your code

**Reply by TangDave** [reply]
Not much code - using a view model as required by DataGrid:

```
		_viewModel = new AvaloniaMVVM.ViewModels.MainWindowHistoryViewModel();
		grdHistory.DataContext = null;
		grdHistory.DataContext = _viewModel;
		grdHistory.Columns[CommonHistory.PreferredHistorySortColumn].Sort(CommonHistory.PreferHistorySortDescending ? ListSortDirection.Descending : ListSortDirection.Ascending);
		grdHistory.SelectedIndex = -1; //why no effect ??

```

**Reply by timunie** [reply, marked as answer]
I'm not used to do it from code behind tbh. You may need to try Dispatcher.UiThread.Post to change selection in order to wait for DataContext and loading is done with all the stuff 🤔 somtimes this helps me

**Reply by TangDave** [reply]
Thanks for the tip - the following worked:
`Avalonia.Threading.Dispatcher.UIThread.Post(() => grdHistory.SelectedIndex = -1);`

Don't quite understand why grdHistory.SelectedIndex = -1 on it's own doesn't work though, but at least it's working.

**Reply by timunie** [reply]
As I said you created it in your constructor. To this time the control isn't loaded correctly and thus some properties may not be setup as needed, for example the ItemsSource. With the dispatcher you can wait for it to finish the heavy liftings.
